tests/unit/dependencies.js

```diff
@@ -0,0 +1,167 @@
+(function () {
```

Comment:
I see what you mean, but the core of the problem is the separation between module dependencies and the injectables they register. Though that problem _is_ compounded by separate files (the issue with most Java code: one file per thing, it takes a long time to find what you want).

E.g. you can have a `angular.module('bar', [])` with `module.factory('foo', ...)`, so you have to depend on the module `bar` to allow injecting the factory `foo`. There's no explicit connection there, only that which you create via naming.

The only advantage I see to Angular 1's module system is avoiding unnecessarily running `config` and `run` blocks that don't end up getting used. All the code is parsed anyway, so it's not really much of a saving.

The latest js frameworks use `import`/`export` so that simply importing something makes it available. That's way simpler and more descriptive.